agent/src/main/java/com/code_intelligence/jazzer/autofuzz/Meta.java

```diff
@@ -0,0 +1,337 @@
+    if (!type.isPrimitive() && data.consumeByte((byte) 0, (byte) 19) == 0) {
+      return null;
+    }
+    if (type.isAssignableFrom(String.class)) {
+      return data.consumeString(data.remainingBytes() / 2);
+    } else if (type.isArray()) {
+      if (type == byte[].class) {
+        return data.consumeBytes(data.remainingBytes() / 2);
+      } else if (type == int[].class) {
+        return data.consumeInts(data.remainingBytes() / 2);
+      } else if (type == short[].class) {
+        return data.consumeShorts(data.remainingBytes() / 2);
+      } else if (type == long[].class) {
+        return data.consumeLongs(data.remainingBytes() / 2);
+      } else if (type == boolean[].class) {
+        return data.consumeBooleans(data.remainingBytes() / 2);
+      } else {
+        Object array = Array.newInstance(type.getComponentType(), data.remainingBytes() / 2);
+        for (int i = 0; i < Array.getLength(array); i++) {
+          Array.set(array, i, consume(data, type.getComponentType()));
+        }
+        return array;
+      }
+    } else if (type.isAssignableFrom(ByteArrayInputStream.class)) {
+      return new ByteArrayInputStream(data.consumeBytes(data.remainingBytes() / 2));
+    } else if (type.isEnum()) {
+      return data.pickValue(type.getEnumConstants());
+    } else if (type == Class.class) {
+      return YourAverageJavaClass.class;
+    } else if (type == Method.class) {
+      return data.pickValue(YourAverageJavaClass.class.getMethods());
+    } else if (type == Constructor.class) {
+      return data.pickValue(YourAverageJavaClass.class.getConstructors());
+    } else if (type.isInterface() || Modifier.isAbstract(type.getModifiers())) {
+      List<Class<?>> implementingClasses = implementingClassesCache.get(type);
+      if (implementingClasses == null) {
+        ClassGraph classGraph =
+            new ClassGraph().enableClassInfo().enableInterClassDependencies().rejectPackages(
+                "jaz.*");
+        if (!isTest()) {
+          classGraph.rejectPackages("com.code_intelligence.jazzer.*");
+        }
+        try (ScanResult result = classGraph.scan()) {
+          ClassInfoList children =
+              type.isInterface() ? result.getClassesImplementing(type) : result.getSubclasses(type);
+          implementingClasses =
+              children.getStandardClasses().filter(cls -> !cls.isAbstract()).loadClasses();
+          implementingClassesCache.put(type, implementingClasses);
+        }
+      }
+      if (implementingClasses.isEmpty()) {
+        if (isDebug()) {
+          throw new AutofuzzConstructionException(String.format(
+              "Could not find classes implementing %s on the classpath", type.getName()));
+        } else {
+          throw new AutofuzzConstructionException();
+        }
+      }
+      return consume(data, data.pickValue(implementingClasses));
+    } else if (type.getConstructors().length > 0) {
+      Constructor<?> constructor = data.pickValue(type.getConstructors());
```

Comment:
Does the constructor list returned by `type.getConstructors()` have the same order in different calls? I had this issue with `type.getMethods()` and had to sort the returned list so that we don't produce bugs that cannot be reproduced with the reported crashing input.

Reply:
Good point, I will sort the output of every method of this kind. I would be surprised if the return value wouldn't be the same across fuzzing runs, but of course it might change when code is recompiled or just across different versions of the JVM.